### PR TITLE
fix(Login): update component to match pf4 design changes

### PR DESCRIPTION
**Pull request by nicolethoen**
Fixes [issue #1239](https://github.com/patternfly/patternfly-react/issues/1239) and [issue #871](https://github.com/patternfly/patternfly-react/issues/871)

screen shot 2019-02-04 at 11 58 27 am
screen shot 2019-02-04 at 11 59 08 am

### CODE REVIEW

**Comment by nicolethoen** [comment]
> Is this PR meant to handle the design changes for the error state? the "invalid login credentials" error message isn't displaying when I don't fill in username/password and press "login".
Otherwise this matches the design specs in Core

I don't believe behaviors are baked into the examples like that, but my understanding could be incorrect. I could do what core does and implement a 'Login Invalid' example alongside the 'Login simple' example.

File: packages/patternfly-4/react-core/src/components/Form/index.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+export { default as ActionGroup } from './ActionGroup';
 export { default as Form } from './Form';
 export { default as FormGroup } from './FormGroup';
-export { default as ActionGroup } from './ActionGroup';
+export { default as FormHelperText } from './FormHelperText';
```
[the remark this answers is not in the document]
Comment (nicolethoen) [reply packages/patternfly-4/react-core/src/components/Form/index.js]: @tlabaj I could be misunderstanding, but LoginForm imports FormHelperText just as it imports Form, FormGroup and ActionGroup. Should I have done that differently?

**Comment by christiemolloy** [comment]
@nicolethoen from my knowledge I believe we would want the error state to be included with this login example, do you know if there is an issue created to handle this? @mcarrano what are your thoughts?

File: packages/patternfly-4/react-core/src/components/Form/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tlabaj) [reply packages/patternfly-4/react-core/src/components/Form/index.js]: You are right.  I miss read this. I see that it is being internally consumed by the LoginForm so I assumed that it was meant to be internal.  Looks good as is.

**Comment by tlabaj** [comment]
@christiemolloy I believe the error state will be covered by this react issue #1009 is that correct @dlabaj?

**Comment by tlabaj** [comment]
@christiemolloy is the error state something different than the Invalid example?

**Comment by mcarrano** [comment]
The question about error states is an interesting one. FWIW, the PF3 React component does bake this into the component. There are a couple of levels of validations that would take place. There is validation that can happen in the client side code, i.e. checking that the user didn't leave the username or password blank, and server side validation that takes place after clicking the submit button. There are different approaches for handling these in the UI. @jknilans has been writing the design doc for this. @nicolethoen perhaps you and Josh can coordinate on what the behavior should be here. IMO it might make sense to bake some common error conditions into the component so they display consistently across product.

In an unrelated issue, how is the Language Selector implemented. It should be a Select list but looks like it behaves as a dropdown.

**Comment by nicolethoen** [comment]
> In an unrelated issue, how is the Language Selector implemented. It should be a Select list but looks like it behaves as a dropdown.

This language selector is using all the same class names - so it seems to be the same sort of component - as the pf4.patternfly.org example. In both instances it is a dropdown as far as I can tell

**Comment by mcarrano** [comment]
OK. That's an error then. I both cases the language selector should be a single-select component (i.e. a custom select).

**Comment by tlabaj** [comment]
@nicole @mcarrano there is an issue to remove the language selector from the Login until the single select is ready.  I put out a PR #1279 to address that but the styling is off with the old Login layout.  I put do not merge in the PR until this one gets merged.  Nicole you are welcome to role that change in here or I can just update my PR once this is merged.

**Pull request merged**